### PR TITLE
wip: removing python 38

### PR DESCRIPTION
**Pull request by iparask**

### CODE REVIEW

**Comment by iparask** [comment]
This PR removes python 3.8 as a requirement for sotodlib. It increments the requirement in setup.py, removes it from the test workflow.

It also removes the deprecated `setuptools.command.test` and updates the test to use `python -m unittest discover`.

**Comment by BrianJKoopman** [comment]
I'm not really a maintainer of this repo (though I am glad to see this) so I'm going to back out of the requested review. If there are any specifics that I might be able to help with feel free to ping me.

**Review by mhasself — APPROVED** [review]
Farewell python 3.8.  I will so miss not being able to use the | operator on dictionaries.

Wait for @tskisner approval on the TestCommand stuff -- I don't know about that...

File: .github/workflows/test.yml
Comment (tskisner) [inline .github/workflows/test.yml:69]: Can we restore the `mpirun` here?  Even 2 processes is enough to catch some problems.

**Review by tskisner — APPROVED** [review]
Looks good, after restoring the MPI tests.

**Pull request merged**
